src/main/java/org/jabref/gui/externalfiles/ImportHandler.java

```diff
@@ -76,6 +76,9 @@ public class ImportHandler {
     private final DialogService dialogService;
     private final TaskExecutor taskExecutor;
 
+    // list is filled as each entry is passed to importCleanedEntries
+    private final List<BibEntry> finalizedEntries = new ArrayList<>();
```

Comment:
rename list to `addedEntries`. `finalized` is a strange word.

Also remove `final` keyword, because the list itself is modified; thus `final` is misleading.